views/012_changelog.sql

```diff
@@ -0,0 +1,75 @@
+-- Trigger def
+CREATE OR REPLACE FUNCTION change_trigger() 
+RETURNS trigger AS $$
+BEGIN
+  IF TG_TABLE_NAME = 'component_relationships' OR TG_TABLE_NAME = 'config_component_relationships' THEN
+    IF TG_OP = 'INSERT' OR TG_OP = 'UPDATE' THEN
+      INSERT INTO push_queue (table_name, operation, item_id) VALUES (TG_RELNAME, TG_OP, NEW.component_id);
+      RETURN NEW;
+    ELSIF TG_OP = 'DELETE' THEN
+      INSERT INTO push_queue (table_name, operation, item_id) VALUES (TG_RELNAME, TG_OP, OLD.component_id);
+      RETURN OLD;
+    END IF;
+  
+  ELSIF TG_TABLE_NAME = 'config_relationships' THEN
+    IF TG_OP = 'INSERT' OR TG_OP = 'UPDATE' THEN
+      INSERT INTO push_queue (table_name, operation, item_id) VALUES (TG_RELNAME, TG_OP, NEW.config_id);
+      RETURN NEW;
+    ELSIF TG_OP = 'DELETE' THEN
+      INSERT INTO push_queue (table_name, operation, item_id) VALUES (TG_RELNAME, TG_OP, OLD.config_id);
+      RETURN OLD;
+    END IF;
+
+  ELSIF TG_TABLE_NAME = 'check_statuses' THEN
+    IF TG_OP = 'INSERT' OR TG_OP = 'UPDATE' THEN
+      INSERT INTO push_queue (table_name, operation, item_id) VALUES (TG_RELNAME, TG_OP, NEW.check_id);
```

Comment:
Check_id will not be unique, it is check_id + time: https://github.com/flanksource/duty/blob/main/schema/checks.hcl#L103